_includes/raw_with_sidebar.tsx

```diff
@@ -0,0 +1,122 @@
+import Logo from "../_components/Logo.tsx";
+import { Sidebar as Sidebar_, SidebarItem } from "../types.ts";
+import { NavigationButton } from "../_components/NavigationButton.tsx";
+import renderCommand from "./renderCommand.tsx";
+
+export const layout = "layout.tsx";
+
+export default function Page(props: Lume.Data, helpers: Lume.Helpers) {
+  const sidebar = props.sidebar as Sidebar_;
+  const file = props.page.sourcePath;
+  if (sidebar === undefined) {
+    throw new Error("Missing sidebar for " + props.url);
+  }
+
+  function walk(
+    sidebarItems: SidebarItem[],
+  ): [SidebarItem[], number] | undefined {
+    for (let i = 0; i < sidebarItems.length; i++) {
+      const sidebarItem = sidebarItems[i];
+      if (typeof sidebarItem === "string") {
+        const data = props.search.data(sidebarItem)!;
+        if (!data) {
+          throw new Error(`No data found for ${sidebarItem}`);
+        }
+
+        if (data.url == props.url) {
+          return [sidebarItems, i];
+        }
+      } else if ("id" in sidebarItem && sidebarItem.id === props.url) {
+        return [sidebarItems, i];
+      } else if ("items" in sidebarItem) {
+        const results = walk(sidebarItem.items);
+        if (results) {
+          return results;
+        }
+      }
+    }
+  }
+
+  let parentNavigation: SidebarItem[] | undefined = undefined;
+  let index: number | undefined = undefined;
+  for (const sidebarElement of sidebar) {
+    const items = walk(sidebarElement.items);
+
+    if (items) {
+      [parentNavigation, index] = items;
+      break;
+    }
+  }
+
+  let renderedCommand = null;
+
+  if (props.command) {
+    const { rendered, toc } = renderCommand(props.command, helpers);
+    renderedCommand = rendered;
+    props.toc = toc.concat(...props.toc);
+  }
+
+  const checkIsHub = /^\/learn\/$/;
+  const isLearnHub = checkIsHub.test(props.url);
+
+  return (
+    <>
+      <aside
+        className={`flex flex-col absolute top-0 xl:top-16 bottom-0 -translate-x-74 xl:left-0 sidebar-open:translate-x-0 w-74 border-r border-foreground-tertiary bg-background-primary z-50 xl:z-0 xl:translate-x-0 transition-transform ${
+          isLearnHub && "xl:hidden"
+        }`}
+        id="sidebar"
+        data-open="false"
+      >
+        <div class="xl:hidden p-4 shadow-sm flex justify-between h-16">
+          <a class="flex items-center gap-3 mr-6" href="/">
+            <Logo />
+          </a>
+          <button
+            type="button"
+            aria-label="Close navigation bar"
+            id="sidebar-close"
+          >
+            <svg
+              viewBox="0 0 15 15"
+              width="16"
+              height="16"
+              class="text-foreground-secondary"
+            >
+              <g stroke="currentColor" stroke-width="1.2">
+                <path d="M.75.75l13.5 13.5M14.25.75L.75 14.25"></path>
+              </g>
+            </svg>
+          </button>
```

Comment:
This works for English, but it's better to use text, rather than an attribute, for localization/i18n purposes. Text is more easily translated, where attributes may not be. So the ideal markup here would look like this (overly simplified) example:

```html
<button>
  <svg aria-hidden="true" />
  <span class="sr-only">Close navigation bar</span>
</button>
```

Reply:
I'm going to raise this as an issue rather than address it in this PR since this is the same sidebar that is used across the site.